processing/src/main/java/io/druid/segment/data/CompressedObjectStrategy.java

```diff
@@ -44,6 +44,8 @@
   private static final Logger log = new Logger(CompressedObjectStrategy.class);
   public static final CompressionStrategy DEFAULT_COMPRESSION_STRATEGY = CompressionStrategy.LZ4;
 
+  // When adding compression strategy do not use id in the range [0x7C, 0xFD], since the id could be subtracted by 126
+  // to indicate whether an encoding format id is needed
```

Comment:
moved all subtracting 126 and compare 0xFE code to CompressionFactory class with some explanation on the flag setting